app/models/pseud.rb

```diff
@@ -116,6 +116,13 @@ def to_param
       ).group('pseuds.id')
   }
 
+  scope :has_works, lambda { |pseud_ids|
+    joins(:works)
+      .where(pseuds: { id: pseud_ids })
```

Comment:
Should this scope have the restriction `hidden_by_admin: false` too? Not a rhetorical question, I just noticed it was present for all other scopes in the file, but I have no idea if it makes sense here or not.